miden-tx/src/tests/kernel_tests/test_account.rs

```diff
@@ -441,6 +449,100 @@ fn test_set_map_item() {
     );
 }
 
+#[test]
+fn test_storage_offset() {
+    // setup assembler
+    let assembler = TransactionKernel::assembler_testing();
+
+    // setup account
+    let id = AccountId::try_from(ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_OFF_CHAIN).unwrap();
+    let vault = AssetVault::mock();
+    let storage = AccountStorage::new(vec![], BTreeMap::new()).unwrap();
+    let source_code = "
+        use.miden::account
+        use.kernel::memory
+
+        export.set_item
+            push.4.5.6.7.0
+            exec.account::set_item
+            dropw dropw
+        end
+
+        export.get_item_0
+            push.0
+            exec.account::get_item
+            swapw dropw swapw
+        end
+
+        export.get_item_1
+            push.1
+            exec.account::get_item
+            swapw dropw
+        end
+    ";
```

Comment:
It took me a while to figure out what these do - so, it may be worth adding comments or maybe refactoring things slightly. My current understanding is that:

- Storage offset for `set_item` is 2. So, `[4, 5, 6, 7]` word gets saved into storage slot 2.
- Storage offset for `get_item_0` is also 2. So, when we do `get_item(0)` we read data from storage slot 2.
- Storage offset for `get_item_1` is 1. So, when we do `get_item(1)` we also read data from storage slot 2.

Then, later we check that both times we've read `[4, 5, 6, 7]` - right?

A potential way to refactor this is to have 4 procedures (and I would probably not name them `set_item`/`get_item` to reduce potential for confusion). The procedures could be:

- `foo_write` and `foo_read` - both with offset 1.
- `bar_write` and `bar_read` - both with offset 2.

Then the test could be to make sure that (a) reads and writes for each set of procedures is consistent and (b) that they actually write into expected slots (for this, we may need to examine the memory of the process).

Reply:
Updated the test and added a comment explaining the logic that we are executing. 

I believe that writing and reading from storage should cover (a) and (b) both for consistency and writing reading to from expected slots without needing to examine memory.

Reply:
For (b), if we could execute this code as a full transaction, then it would be pretty straight-forward as we'd be able to see the final storage state.

Without that, you could probably read the storage commitment from the process memory and check if it matches the expected storage commitment. This also shouldn't be too difficult.

Reply:
Does [this work/help](https://github.com/0xPolygonMiden/miden-base/pull/853)? It includes a small refactor for builder structs and cleans up the test a bit.

Reply:
Looks good @igamigo, I have pushed a PR to your branch updating the test to also assert that we correctly "get" elements from storage with the right offset.

Though I am having some LSP errors on your branch that I have outlined in a comment on the PR, let me know if you are also having them and if so we should fix before merging.

Reply:
> I talked to @phklive a couple hours ago and told him basically this. In the next few moments I'll work on sending an example over this test and see if there is anything else that should be added to make this simpler as well (like the account code mocking)

Yes, your PR seems to fix what we we're talking about, as we also discussed it would be difficult to go away from arbitrary code execution inside the kernel to test some isolated kernel functions.

Should we permeate the changes to the other tests?